implementation/src/main/java/io/smallrye/config/KeyMapBackedConfigSource.java

```diff
@@ -0,0 +1,33 @@
+package io.smallrye.config;
+
+import java.util.Map;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+import io.smallrye.config.common.AbstractConfigSource;
+
+public abstract class KeyMapBackedConfigSource extends AbstractConfigSource {
+    private static final long serialVersionUID = 4378754290346888762L;
+
+    private final KeyMap<String> properties;
+
+    public KeyMapBackedConfigSource(final String name, final KeyMap<String> properties) {
+        super(name, ConfigSource.DEFAULT_ORDINAL);
+        this.properties = properties;
+    }
+
+    public KeyMapBackedConfigSource(final String name, final int ordinal, final KeyMap<String> properties) {
+        super(name, ordinal);
+        this.properties = properties;
+    }
+
+    @Override
+    public Map<String, String> getProperties() {
+        return properties.toMap();
```

Comment:
This is still not correct.  The "toMap" method on keyMap is not meaningful in this way.  KeyMap is pattern-based, not key-value based.  Therefore `getProperties()` can never be meaningful, and `toMap()` on that method probably should not exist (it is only going to be misused).